lib/coupon_code.rb

```diff
@@ -21,8 +21,8 @@ def self.generate(options = { parts: PARTS })
 
   def self.validate(orig, num_parts = PARTS)
     code = orig.upcase
-    code.gsub!(/[^0-9A-Z]+/, '')
-    parts = code.scan(/[0-9A-Z]{#{LENGTH}}/)
+    code.gsub!(reg_symbol, '')
```

Comment:
Hi @mrclmrvn, thanks for the feedback. I really appreciate your time and effort.

I think replacing `reg_symbol` with a regular expression doesn't add more lines.

What I'm suggesting is:
````ruby
code.gsub!(/[^#{SYMBOL}]/, '')
parts = code.scan(/^[#{SYMBOL}]{#{LENGTH}}$/)
````
and I think the complexity it adds is reasonable.

(Just a couple notes, since you used `#gsub!`, `+` modifier is not necessary for the first regex and for the second regex, by specifying `^...$` we can filter forbid strings that are longer than `LENGTH`.)

Adding two more methods concerns me as they are:
 - publicly accessible while they don't need to be known to the users of this gem.
 - not actually re-used so no significant benefit from having them as a method.

What do you think?